python-lib/nlp_utils.py

```diff
@@ -3,16 +3,37 @@
 import unicodedata
 
 
-def get_keyword(text: AnyStr, normalize_case: bool) -> AnyStr:
+def get_text_case(text: AnyStr, normalize_case: bool) -> AnyStr:
     """Return text in its wanted-case form"""
     return text.lower() if normalize_case else text
 
+def get_attribute(lemmatize: bool) -> AnyStr:
+    """Return the right attribute to pass to spaCy Matcher"""
+    return "LEMMA" if lemmatize else "TEXT"
 
-def get_sentence(span: Span, normalize_case: bool) -> Union[Span, Doc]:
-    """Return Span object as a Doc if case_insensitive is set to True"""
-    return span if normalize_case else span.as_doc()
+
+def get_text_normalized(text: Doc, normalize_case: bool) -> AnyStr:
+    """Lemmatize text and return it lowercased if normalize_case is True"""
+    text = " ".join([word.lemma_ for word in text])
+    return get_text_case(text, normalize_case)
+
+
+def get_keyword(normalize_case: bool, lemmatize: bool, keyword: Span) -> AnyStr:
+    """Return text after normalizing it if needed
```

Comment:
Which normalization? It's important to use explicit terms and to explain what a function does in layman's terms.

Reply:
0765d3c